src/execution_plan/execution_plan.c

```diff
@@ -314,11 +314,17 @@ static void _ExecutionPlan_ProcessQueryGraph(ExecutionPlan *plan, QueryGraph *qg
 				 * try to locate and remove it, there's no real harm except some performace hit
 				 * in keeping that label matrix. */
 				if(exp->operands[0].diagonal) AlgebraicExpression_RemoveTerm(exp, 0, NULL);
-				tail = NewNodeByLabelScanOp(plan, src);
+				root = tail = NewNodeByLabelScanOp(plan, src);
 			} else {
-				tail = NewAllNodeScanOp(plan, gc->g, src);
+				root = tail = NewAllNodeScanOp(plan, gc->g, src);
 			}
 
+			/* The expression has been fully converted - the QueryGraph had edges, but we don't
+			 * need to build traversals. This can occur in cases like:
+			 * MATCH (a)-[*0]->(c) RETURN c
+			 * We can just mark the destination alias as referring to the same entity as the source. */
+			if(exp->operand_count == 0) OpBase_AliasModifier(root, exp->src, exp->dest);
```

Comment:
In the example: 
`MATCH (a)-[*0]->(c) RETURN c` I'm expecting an algebraic expression with a single operand: THE adjacency matrix, although the requirement is 0 hops, because neither the source node and the destination node have labels, an AllNodeScan operation will be used to resolve one of them, this leaves `exp->operand_count` to be greater than 0.

I prefer we avoid calling `OpBase_AliasModifier` from execution_plan.c as this function should be called from within an operation "constructor". 

I think we should try address 0 length traversals at the AlgebraicExpression level, maybe completely discarding the traversal and marking the operand M (if there's one, if not introduce it) source and destination aliases to `a` and `c`.

Reply:
The example actually constructs an algebraic expression with 0 operands, because loops over hops evaluate 0 times - https://github.com/RedisGraph/RedisGraph/blob/master/src/arithmetic/algebraic_expression.c#L487-L491 .

I agree that this would be better handled at the AlgebraicExpression level, but I don't want to waste too much time trying to augment the logic there if we're going to replace it soon. I don't feel like this fix is worth that much time, really, and adding new conditional logic to AlgebraicExpression building always seems precarious. How much time do we think is worth spending on this?